testcases/kernel/syscalls/fanotify/fanotify11.c

```diff
@@ -61,16 +61,11 @@ void test01(unsigned int i)
 	int fd_notify;
 	int tgid = getpid();
 
-	fd_notify = fanotify_init(tcases[i], 0);
-	if (fd_notify < 0) {
-		if (errno == EINVAL && (tcases[i] & FAN_REPORT_TID)) {
-			tst_res(TCONF,
-				"FAN_REPORT_TID not supported in kernel?");
-			return;
-		}
-		tst_brk(TBROK | TERRNO, "fanotify_init(0x%x, 0) failed",
-				tcases[i]);
-	}
+	tst_res(TINFO, "Test #%u: %s FAN_REPORT_TID: tgid=%d, tid=%d, event.pid=%d",
+			i, (tcases[i] & FAN_REPORT_TID) ? "with" : "without",
+			tgid, tid, event.pid);
+
+	fd_notify = SAFE_FANOTIFY_INIT(tcases[i], 0);
 
```

Comment:
Actually this changes how the test behaves considerably. In the original code if fanotify_init() with FAN_REPORT_TID reported EINVAL the case was skipped, but the test carried on. After the change the test will be aborted immediately. Which means that if the test is passed -i parameter on a system that does not support FAN_REPORT_TID it will do only single iteration instead.

Reply:
Or better, we can check for fanotify support in the test setup with:

```
fd = SAFE_FANOTIFY_INIT(...);
SAFE_CLOSE(fd); 
```

Reply:
@rantala we are about to produce a LTP release and I would like to have this fix in. Will you update the patch or should I do it?

Reply:
> @rantala we are about to produce a LTP release and I would like to have this fix in. Will you update the patch or should I do it?

Great, I'll try to update this patch still today.